components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -365,24 +374,41 @@ protected void buildClaims(AuthenticationContext context, Map<String, Object> js
                     }
                 }
             }
-
-            if (StringUtils.isBlank(context.getExternalIdP().getIdentityProvider()
-                    .getClaimConfig().getUserClaimURI())) {
-                context.getExternalIdP().getIdentityProvider().getClaimConfig().setUserClaimURI
-                        (FacebookAuthenticatorConstants.EMAIL);
-            }
-            String subjectFromClaims = FrameworkUtils.getFederatedSubjectFromClaims(
-                    context.getExternalIdP().getIdentityProvider(), claims);
-            if (subjectFromClaims != null && !subjectFromClaims.isEmpty()) {
-                AuthenticatedUser authenticatedUser =
-                        AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(subjectFromClaims);
-                context.setSubject(authenticatedUser);
-            } else {
-                setSubject(context, jsonObject);
+            if (claimConfig != null) {
+                if (StringUtils.isBlank(claimConfig.getUserClaimURI())) {
+                    claimConfig.setUserClaimURI(getEffectiveClaimUri(getClaimDialectURI(),
+                            FacebookAuthenticatorConstants.EMAIL));
+                }
+                String subjectFromClaims = null;
+                try {
+                    if (StringUtils.isNotBlank(claimConfig.getUserClaimURI()) && getClaimDialectURI() != null ) {
```

Comment:
can use either StringUtils isNotEmpty or isNotBlank for the second check as well.

Reply:
done